__tests__/server.test.js

```diff
@@ -12,8 +12,29 @@ const seed = require("../db/seeds/seed");
 beforeEach(() => seed({ articleData, commentData, topicData, userData }));
 afterAll(() => db.end());
 
+describe("* ALL incorrect endpoints", () => {
+    test("404 - sends an error message to the client when a request is made to a path that does not exist", () => {
+        return request(app)
+            .get("/api/notvalid")
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe("ERROR: path not found");
+            });
+    });
+});
+
 describe("API endpoints", () => {
-    describe("", () => {
-        it("", () => {});
+    describe("GET: /api/topics ", () => {
+        test("200 - sends an array of topics to the client", () => {
+            return request(app)
+                .get("/api/topics")
+                .expect(200)
+                .then((response) => {
+                    response.body.topics.forEach((topic) => {
```

Comment:
Because an array could be empty and if it is so this forEach will never run and thus the expects never run you need to assert the length of the array before you run the forEach (in the same test)

Reply:
Ah yes thanks Mark!